src/catchup/ApplyBucketsWork.cpp

```diff
@@ -111,7 +111,7 @@ ApplyBucketsWork::doReset()
             if (bucket->getSize() > 0)
             {
                 mTotalBuckets++;
-                mTotalSize += bucket->getSize();
+                mTotalSize = saturatingAdd(mTotalSize, bucket->getSize());
```

Comment:
I think this change adds confusion rather than clarity, because bucket size will never be close to uint_max (they aren't user-configurable)

Reply:
Sure, here is accumulating all buckets sizes into a total, would be somewhat large I guess (not close to a uint64 max but could overflow uint32)? 
Anyway, I don't feel super strong about this, just trying to be more defensive (coming from Rust side where `checked_` arith is baked in and often preferred over raw arith).